docs/contributor/arch/015-impact-of-istio-prometheus-merge-on-metric-pipelines.md

```diff
@@ -0,0 +1,42 @@
+# 11. Impact of Istio Prometheus Merge Feature on MetricPipelines
+
+Date: 2024-10-21
+
+## Status
+
+Proposed
+
+## Context
+
+The MetricPipeline supports various input types, including `prometheus` and `istio`.
+* When the `prometheus` input is enabled, it scrapes metrics from Pods and Services annotated with "prometheus.io/scrape=true".
+* When the `istio` input is enabled, it scrapes metrics from the Istio proxy (sidecar) containers injected into Pods.
+
+Both inputs are technically backed by the OpenTelemetry Collector’s Prometheus Receiver. There are five distinct scrape jobs, each targeting different configurations:
+
+| Scrape Job | Targets |
+| --- | --- |
+| app-pods | Discovers annotated Pods that either do not have an Istio sidecar or have the "prometheus.io/scheme" annotation set to "http". Scraping is done over plain HTTP. |
+| app-services | Discovers annotated Services backed by Pods without an Istio sidecar or have the "prometheus.io/scheme" Service annotation set to "http". Scraping is done over plain HTTP. |
+| app-pods-secure | Discovers annotated Pods that either have an Istio sidecar or have the "prometheus.io/scheme" annotation set to "https". Scraping is performed over HTTPS, using a client TLS certificate injected by Istio. |
+| app-services-secure | Discovers annotated Services backed by Pods with an Istio sidecar or where the "prometheus.io/scheme" Service annotation is set to "https". Scraping is done over HTTPS, using a client TLS certificate injected by Istio. |
+| istio-proxy | Discovers metrics from Istio proxy sidecars. Scraping is done over plain HTTP. |
+
+In general, annotating Services offers much more flexibility than annotating Pods due to the following reasons:
+* A Pod can only have a single "prometheus.io/port" annotation, meaning that if multiple containers are running in the Pod (or single container exposes multiple ports), only one can be scraped. With Services, multiple Services can be created to target different ports, allowing all relevant metrics to be scraped.
+* For istiofied Pods, even with annotations, a Service is still required to define the application protocol in order for scraping to occur. The Service itself doesn’t need annotations but must be present to enable scraping. See more: https://kyma-project.io/#/telemetry-manager/user/04-metrics?id=_4-activate-prometheus-based-metrics.
+
+As discussed in [this issue](https://github.com/kyma-project/telemetry-manager/issues/1468), the `prometheusMerge` feature is crucial for simplifying Dynatrace integration and enhancing security. For the sake of explanation, let’s assume that `prometheusMerge` is always enabled in the Istio mesh configuration. Under this scenario, the following points are true:
```

Comment:
Just in case the links breaks, this makes it easier for readers to find the ticket on their own.

```suggestion
As discussed in [issue #1468](https://github.com/kyma-project/telemetry-manager/issues/1468), the `prometheusMerge` feature is crucial for simplifying Dynatrace integration and enhancing security. For the sake of explanation, let’s assume that `prometheusMerge` is always enabled in the Istio mesh configuration. Under this scenario, the following points are true:
```